frontend/src/metabase/query_builder/components/template_tags/CardTagEditor.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router";
+import { t } from "ttag";
+
+import Icon from "metabase/components/Icon";
+import Card from "metabase/components/Card";
+import QuestionPicker from "metabase/containers/QuestionPicker";
+import PopoverWithTrigger from "metabase/components/PopoverWithTrigger";
+import SelectButton from "metabase/components/SelectButton";
+import LoadingSpinner from "metabase/components/LoadingSpinner";
+
+import Questions from "metabase/entities/questions";
+import { question as questionUrl } from "metabase/lib/urls";
+import { formatDateTimeWithUnit } from "metabase/lib/formatting";
+import MetabaseSettings from "metabase/lib/settings";
+
+function mapStateToProps(state, props) {
```

Comment:
Unless I'm missing something, I think this could all be replaced with the question entity loader (the prop will be named `question` instead of `card`, though)

```
import Question from "metabase/entities/questions";

@Questions.load({ id: (state, { tag }) => tag.card })
```

Actually, it looks like that would conflict with the `question` prop already being passed in. I still think it would be better to figure out how to use the Question entity loader though.

One idea would be adding a prop to `EntityObjectLoader` (and `EntityListLoader`) to allow you to pick the alias which overrides the default (`entityDef.nameOne`):

```
@Questions.load({
  id: ...,
  alias: "card"
})
```

replacing

```
    return children({
      ..._.omit(props, ...CONSUMED_PROPS),
      object,
      // alias the entities name:
      [entityDef.nameOne]: object,
      reload: this.reload,
      remove: this.remove,
    });
```

with 

```
    return children({
      ..._.omit(props, ...CONSUMED_PROPS),
      object,
      // alias the entities name:
      [props.alias || entityDef.nameOne]: object,
      reload: this.reload,
      remove: this.remove,
    });
```

Or maybe a generic way of picking/renaming parameters:

```
@Questions.load({
  id: ...,
  props: ({ object }) => ({ card: object })
})
```

Either way make sure to add the new prop to `CONSUMED_PROPS`

Reply:
I had started with the entity loader but I removed it to customize the 404 behavior. Though, now I remember there's a flag to suppress that anyways. I'll swap it back in.